### DISCUSSION TITLE
should local scope builder function should return the $query or not?

### DISCUSSION THREAD

**Discussion by RoniNes** [laravel/framework#43614] [Q&A]
Hi,
In [Local Scopes](https://laravel.com/docs/9.x/eloquent#local-scopes) doc, one function returns the `$query`, the other, not.

```
    public function scopePopular($query)
    {
        return $query->where('votes', '>', 100);
    }

    public function scopeActive($query)
    {
        $query->where('active', 1);
    }
```

which is correct?
should any scope function return `$query`

**Comment by henzeb** [comment]
It doesn't really matter (just tested it). the scope method is handled by magic method and then the Builder instance is returned to you even if you did not return it inside.

**Comment by michael-rubel** [comment]
Not necessarily. You can use return or void, but if you want better testing in isolation I'd recommend using types and returning.
